src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIBookingService.kt

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.CommunityAPIClient
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ActionPlanAppointment
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import javax.validation.constraints.NotNull
+
+@Service
+class CommunityAPIBookingService(
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.view-appointment}") private val interventionsUIViewAppointment: String,
+  @Value("\${community-api.locations.book-appointment}") private val communityApiBookAppointmentLocation: String,
+  @Value("\${community-api.crs.office-location}") private val crsOfficeLocation: String,
+  @Value("\${community-api.crs.bookings-context}") private val crsBookingsContext: String,
+  val communityAPIClient: CommunityAPIClient,
+) {
+
+  fun book(existingAppointment: ActionPlanAppointment, appointmentTime: OffsetDateTime?, durationInMinutes: Int?) {
+    when {
+      isInitialBooking(existingAppointment, appointmentTime, durationInMinutes) -> {
+        makeInitialBooking(existingAppointment, appointmentTime, durationInMinutes)
+      }
+      else -> {}
+    }
+  }
+
+  private fun makeInitialBooking(appointment: ActionPlanAppointment, appointmentTime: OffsetDateTime?, durationInMinutes: Int?) {
+    val resourceUrl = UriComponentsBuilder.fromHttpUrl(interventionsUIBaseURL)
+      .path(interventionsUIViewAppointment)
+      .buildAndExpand(appointment.actionPlan.referral.id)
+      .toString()
+
+    val appointmentCreateRequestDTO = AppointmentCreateRequestDTO(
+      appointmentDate = appointmentTime!!.toLocalDate(),
```

Comment:
why is appointment time nullable in the method signature?

Reply:
in that case we can do the not null assertion at the call site, and make this method signature reflect the expections that the params are not null.

Reply:
i.e.

```
isInitialBooking(existingAppointment, appointmentTime, durationInMinutes) -> {
            makeInitialBooking(existingAppointment, appointmentTime!!, durationInMinutes!!)
          }
          ```

Reply:
Done